app/src/main/java/com/github/fribourgsdp/radio/data/view/AddPlaylistActivity.kt

```diff
@@ -128,23 +125,30 @@ open class AddPlaylistActivity : DatabaseHolder, MyAppCompatActivity(), SavePlay
 
 
     override fun onPick(online: Boolean) {
-        val playlistName : String = findViewById<EditText>(R.id.newPlaylistName).text.toString()
-        val genre : Genre = genreSpinner.selectedItem as Genre
-        val playlist = Playlist(playlistName,listSongs.toSet(),genre)
-        val playlistTask = if (online) playlist.transformToOnline().addOnSuccessListener {
-            playlist.saveOnline()
-        } else Tasks.forResult(playlist)
-
-
-        playlistTask.addOnSuccessListener {
-            user.addPlaylist(playlist)
-            user.save(this)
-            user.onlineCopyAndSave()
-            val intent = Intent(this@AddPlaylistActivity, UserProfileActivity::class.java)
-            startActivity(intent)
+        val playlistName: String = findViewById<EditText>(R.id.newPlaylistName).text.toString()
+        val genre: Genre = genreSpinner.selectedItem as Genre
+        var playlist = Playlist(playlistName, listSongs.toSet(), genre)
+        val t = if (generateLyricsCheckBox.isChecked) {
+            playlist.loadLyrics()
+            //loadLyrics(playlist)
+        } else {
+            Tasks.forResult(null)
+        }
+        t.continueWith {
+            //playlist = it.result
+            val playlistTask = if (online) playlist.transformToOnline().addOnSuccessListener {
+                playlist.saveOnline()
```

Comment:
I've just seen that saveOnline() calls db.registerSong() for every song in the playlist, which returns a Task<Void> but does not wait for the result, wouldn't this cause a problem if db.registerPlaylist(this) finishes before a song has finished being registered? (I know this is not from this weeks PR but what do you think?)

Reply:
Yes you are right, I corrected it in the Playlist.kt file.